internal/server/models/identity.go

```diff
@@ -19,7 +21,9 @@ type Identity struct {
 	LastSeenAt time.Time // updated on when an identity uses a session token
 	CreatedBy  uid.ID
 
-	Groups []Group `gorm:"many2many:identities_groups"`
+	// for eager loading, don't use these for saving.
+	Groups    []Group    `gorm:"many2many:identities_groups"`
+	Providers []Provider `gorm:"many2many:provider_users;"`
```

Comment:
Should this relation be on provider users?

Reply:
I kind of hate these relations. definitely don't want to add them anywhere they're not needed. I'm disturbed by the [gorm behavior](https://gorm.io/docs/associations.html) of saving associations automatically, which seems like a recipe for disaster.